File: app/src/main/java/temple/edu/yelp_randomizer/models/RestaurantModel.java
package temple.edu.yelp_randomizer.models;

import android.os.Parcel;
import android.os.Parcelable;

public class RestaurantModel implements Parcelable {
    private int rating;
    private String id; // not accessible to user
    private String name; // name
    private String phone; // phone
    private String image; // image_url
    private String location; // location has a JSON array
    private String url;

    public RestaurantModel(int rating, String id, String name, String phone, String image, String location, String url) {
        this.rating = rating;
        this.id = id;
        this.name = name;
        this.phone = phone;
        this.image = image;
        this.location = location;
        this.url = url;
    }

    protected RestaurantModel(Parcel in) {
        id = in.readString();
        name = in.readString();
        phone = in.readString();
        image = in.readString();
        location = in.readString();
        url = in.readString();
    }

    public static final Creator<RestaurantModel> CREATOR = new Creator<RestaurantModel>() {
        @Override
        public RestaurantModel createFromParcel(Parcel in) {
            return new RestaurantModel(in);
        }

        @Override
        public RestaurantModel[] newArray(int size) {
            return new RestaurantModel[size];
        }
    };

    public int getRating() {
        return rating;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(id);
        parcel.writeString(name);
        parcel.writeString(phone);
        parcel.writeString(image);
        parcel.writeString(location);
        parcel.writeString(url);
    }
}

File: app/src/main/java/temple/edu/random/models/RestaurantModel.kt
package temple.edu.random.models

import android.os.Parcelable
import android.os.Parcel
import android.os.Parcelable.Creator

class RestaurantModel : Parcelable {
    var rating = 0
    var id // not accessible to user
            : String?
    var name // name
            : String?
    var phone // phone
            : String?
    var image // image_url
            : String?
    var location // location has a JSON array
            : String?
    var url: String?

    constructor(rating: Int, id: String?, name: String?, phone: String?, image: String?, location: String?, url: String?) {
        this.rating = rating
        this.id = id
        this.name = name
        this.phone = phone
        this.image = image
        this.location = location
        this.url = url
    }

    protected constructor(`in`: Parcel) {
        id = `in`.readString()
        name = `in`.readString()
        phone = `in`.readString()
        image = `in`.readString()
        location = `in`.readString()
        url = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(id)
        parcel.writeString(name)
        parcel.writeString(phone)
        parcel.writeString(image)
        parcel.writeString(location)
        parcel.writeString(url)
    }

    companion object {
        val CREATOR: Creator<RestaurantModel?> = object : Creator<RestaurantModel?> {
            override fun createFromParcel(`in`: Parcel): RestaurantModel? {
                return RestaurantModel(`in`)
            }

            override fun newArray(size: Int): Array<RestaurantModel?> {
                return arrayOfNulls(size)
            }
        }
    }
}